lib/llm-events/openai/llm-error.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+/**
+ * Represents an error object, to be tracked via `agent.errors`, that is the
+ * result of some error returned from the OpenAI API.
+ */
+class OpenAiLlmError extends Error {
```

Comment:
i'll review more at length later but wanted to call out we already had openai/error-message.js.  That class should be pared down but you can see there are no complex objects. when dealing with nested keys they are just `.` delimited like `error.code` or `http.statusCode`

Reply:
What is the underlying reasoning behind `obj['error.code']` style? My guess is that it is something around data transmission and storage? Just wanting to understand.



> we already had openai/error-message.js

Should I update that object with the extra properties required by the spec? Is the spec defining the minimum properties for the error, or is it defining the whole set of properties?

Reply:
we spoke in slack about this